src/ert/_c_wrappers/enkf/res_config.py

```diff
@@ -265,6 +265,7 @@ def _alloc_from_dict(self, config_dict):
         )
         jobs = []
         for job_name, args in config_dict.get(ConfigKeys.FORWARD_MODEL, []):
+            job_name = self.substitution_list.substitute(job_name)
```

Comment:
would it be reasonable to use a different name for the variable holding the substituted job name?